### PR TITLE
Knock explicit struct

### PR DESCRIPTION
**Pull request by rusefillc**

### CODE REVIEW

**Comment by mck1117** [comment]
🤔🤔🤔🤔🤔

image

**Comment by rusefillc** [comment]
> 🤔🤔🤔🤔🤔
> 
> image

I know https://github.com/FOME-Tech/fome-fw/issues/346

File: firmware/controllers/sensors/impl/software_knock.cpp [hunk 1]
```diff
@@ -55,38 +55,46 @@ static const uint32_t smpr2 =
...
-
-	0,	// htr
-	0,	// ltr
-
-	0,	// sqr1
-	0,	// sqr2
-	ADC_SQR3_SQ1_N(KNOCK_ADC_CH2)
+static const ADCConversionGroup adcConvGroupCh2 = {
+    .circular = FALSE,
+   	.num_channels = 1,
```
Comment (mck1117) [inline firmware/controllers/sensors/impl/software_knock.cpp:82]: strange indent?

Comment (rusefillc) [reply firmware/controllers/sensors/impl/software_knock.cpp:82]: @mck1117 should be better now

Comment (mck1117) [reply firmware/controllers/sensors/impl/software_knock.cpp:82]: ok, indents were _not_ better, something is very wrong with your editor

image

Comment (rusefillc) [reply firmware/controllers/sensors/impl/software_knock.cpp:82]: > ok, indents were _not_ better, something is very wrong with your editor

my editor might be missing an .editorconfig in this repo?

Comment (mck1117) [reply firmware/controllers/sensors/impl/software_knock.cpp:82]: > my editor might be missing an .editorconfig in this repo?

There are lots of funny spots in rusEFI too.

image

**Pull request merged**